lib/middleware/fetch-open-profile-tasks.js

```diff
@@ -0,0 +1,19 @@
+
+module.exports = id => (req, res, next) => {
+  if (!req.establishment || !req.profileId) {
+    return next();
+  }
+
+  return req.workflow.profileTasks(req.profileId, req.establishment.id)
+    .then(workflowResponse => {
+      res.meta = res.meta || {};
+      res.meta.openTasks = workflowResponse.json.data || [];
+      next();
+    })
+    .catch(error => {
+      req.log('error', { ...error, stack: error.stack, message: error.message });
+      res.meta = res.meta || {};
+      res.meta.openTasks = [{ action: 'failed' }];
```

Comment:
I'm guessing this is copied from the other "get tasks" middleware and is to stop stuff being incorrectly allowed if the task loading fails?

Reply:
Yeah it's basically a copy of https://github.com/UKHomeOffice/asl-public-api/blob/master/lib/middleware/fetch-open-tasks.js with some slight tweaks

Reply:
Not actually sure what that's for but thought it best to copy the response, as I'm assuming it must be used somewhere.

Reply:
I believe it's because there's a lot of code that checks for the existence of open tasks to determine if a thing can be edited. We don't want to allow things to be edited if the task lookup fails.

Reply:
(but also don't want to completely fail to display a read-only version if workflow fails)